src/metldata/builtin_transformations/common/assumptions.py

```diff
@@ -0,0 +1,160 @@
+            f"does not describe a direct relation, but contains {
+                num_elements} different relations"
+        )
+
+
+def assert_class_is_source(*, path: RelationPath, instruction: InstructionProtocol):
+    """Ensure that the class being modified with the reference count property is the expected class.
+    This function iterates over the elements of the relation path in the given instruction
+    and validates that the class being modified with the reference count property matches
+    the class specified in the relation path.
+    """
+    for path_element in path.elements:
+        _validate_modification_class(path_element, instruction.class_name)
+
+
+def _validate_modification_class(
+    path_element: RelationPathElement, expected_class_name: str
+):
+    """Check whether the class specified to be modified with the reference count
+    matches the source or target class in the provided `path_element`, depending on the
+    type of the relation path (i.e., active or passive). If the class does not match,
+    an exception is raised.
+    """
+    modification_class_name = (
+        path_element.source
+        if path_element.type_ == RelationPathElementType.ACTIVE
+        else path_element.target
+    )
+    if expected_class_name != modification_class_name:
+        raise ModelAssumptionError(
+            f"Class {
+                expected_class_name} does not correspond to the relation source "
+            f"{modification_class_name}."
+        )
+
+
+def assert_object_path_exists(
+    *,
+    model: SchemaPack,
+    instruction: InstructionProtocol,
+) -> None:
+    """Make sure that the source class (the class being modified) and the object_path
+    exists in the model. Assumption fails if the content path is in the model before
+    the transformation.
+    """
+    class_name = instruction.class_name
+    class_def = model.classes.get(class_name)
+
+    # Check if the class exists in the model
+    if not class_def:
+        raise ModelAssumptionError(f"Class {class_name} does not exist in the model.")
+
+    # Check if the object_path already exists in the model
+    try:
+        target_schema = resolve_schema_object_path(
+            json_schema=class_def.content.json_schema_dict,
+            path=instruction.target_content.object_path,
+        )
+    except KeyError as exc:
+        raise ModelAssumptionError(
+            f"Object path {
+                instruction.target_content.object_path} does not exist"
+            + f" in class {class_name}."
+        ) from exc
+
+    # Check if the property_name already exists in the model and raise an error if so
+    if instruction.target_content.property_name in target_schema.get("properties", {}):
+        raise ModelAssumptionError(
+            f"Property {
+                instruction.target_content.property_name} already exists"
+            + f" in class {class_name}."
+        )
+
+
+def assert_multiplicity(*, model: SchemaPack, path: RelationPath):
```

Comment:
Missing return type.
Also, function name could be a bit more specific as this is only checking target multiplicity.